tests/ui/type-alias-impl-trait/issue-96572-unconstrained.rs

```diff
@@ -63,6 +61,19 @@ mod only_pattern {
             None => {}
         }
     }
+
+    type V = impl Copy;
+
+    fn baz(baz: Option<V>) {
+        match baz {
+            _ => {}
+            Some((mut x, mut y)) => {
+                x = 42;
+                y = "foo";
+            }
+            None => {}
+        }
+    }
```

Comment:
What is this testing? It works on nightly already.

Reply:
That caused an ICE in the one-pass rewrite I'm working on, I thought I'd add it defensively

Reply:
It's a case where the first pattern also has an opaque type, so looking at the first pattern can't save us. I think it ICEs one of my other open PRs even